osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -90,26 +89,33 @@ protected override void LoadComplete()
                 settings => settings.Offset,
                 val => userBeatmapOffsetClock.Offset = val);
 
-            // sane default provided by ruleset.
-            startOffset = gameplayStartTime;
-
-            if (!startAtGameplayStart)
+            if (GameplayStartTime == null)
             {
-                startOffset = Math.Min(0, startOffset);
-
-                // if a storyboard is present, it may dictate the appropriate start time by having events in negative time space.
-                // this is commonly used to display an intro before the audio track start.
-                double? firstStoryboardEvent = beatmap.Storyboard.EarliestEventTime;
-                if (firstStoryboardEvent != null)
-                    startOffset = Math.Min(startOffset, firstStoryboardEvent.Value);
-
-                // some beatmaps specify a current lead-in time which should be used instead of the ruleset-provided value when available.
-                // this is not available as an option in the live editor but can still be applied via .osu editing.
-                if (beatmap.BeatmapInfo.AudioLeadIn > 0)
-                    startOffset = Math.Min(startOffset, firstHitObjectTime - beatmap.BeatmapInfo.AudioLeadIn);
-            }
+                // sane default provided by ruleset.
+                double offset = gameplayStartTime;
```

Comment:
Would rename `gameplayStartTime` to not conflict with the introduced property `GameplayStartTime` (maybe `rulesetGameplayStartTime`).

Reply:
Wow, I didn't even see this variable conflict.